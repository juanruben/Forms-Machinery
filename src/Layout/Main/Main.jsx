import React from 'react';
import PropTypes from 'prop-types';
import { Route, Switch, withRouter } from 'react-router-dom';
import Sidebar from 'react-sidebar';
import { StateContext } from '../../State';
import Menu from '../../Components/Menu/Menu';
import MenuButton from '../../Components/MenuButton/MenuButton';
import Title from '../../Components/Title/Title';
import History from '../../Pages/History/History';
import Clients from '../../Pages/Clients/Clients';
import Machines from '../../Pages/Machines/Machines';
import Users from '../../Pages/Users/Users';
import Forms from '../../Pages/Forms/Forms';
import Sections from '../../Pages/Forms/Sections';
import Fields from '../../Pages/Forms/Fields';
import CheckIn from '../../Pages/CheckIn/CheckIn';
import CheckOut from '../../Pages/CheckOut/CheckOut';

import menuAdmin from './data/dataMenuAdmin';
import menuOp from './data/dataMenuOp';

import avatar from './avatar.png';

import './Main.scss';

const mql = window.matchMedia('(min-width: 800px)');

class Main extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            docked: mql.matches,
            transitions: false,
            dataMenu: [],
            username: '',
        };

        this.menuButtonClick = this.menuButtonClick.bind(this);
        this.mediaQueryChanged = this.mediaQueryChanged.bind(this);
        this.checkLoggedIn = this.checkLoggedIn.bind(this);
        this.findTitle = this.findTitle.bind(this);
    }

    componentDidMount() {
        mql.addListener(this.mediaQueryChanged);
        this.findTitle();
        this.checkLoggedIn();
    }

    componentWillUnmount() {
        mql.removeListener(this.mediaQueryChanged);
    }

    mediaQueryChanged() {
        this.setState({
            docked: mql.matches,
        });
    }

    checkLoggedIn() {
        const [{ role }, dispatch] = this.context;
        if (!role) {
            dispatch({
                type: 'LOAD_SESSION',
            });
        } else {
            this.setState({
                dataMenu: role === 1 ? menuAdmin : menuOp,
                username: role === 1 ? 'administrador' : 'operador',
            }, () => {
                this.checkPermission();
            });
        }
    }

    checkPermission() {
        const { location } = this.props;
        const { pathname } = location;
        const { dataMenu } = this.state;
        const [, dispatch] = this.context;

        const item = dataMenu.find((v) => v.path === pathname);

        if (!item) {
            dispatch({
                type: 'EXIT',
            });
        }
    }

    findTitle() {
        const { location } = this.props;
        const { pathname } = location;
        const { dataMenu } = this.state;

        const item = dataMenu.find((v) => v.path === pathname);
        return item ? item.title : '';
    }

    menuButtonClick(event) {
        event.preventDefault();
        const { docked } = this.state;
        this.setState({
            transitions: true,
            docked: !docked,
        });
    }

    render() {
        const {
            docked, transitions, dataMenu, username,
        } = this.state;

        const sidebarProps = {
            sidebar: <Menu items={dataMenu} username={username} avatar={avatar} />,
            docked,
            shadow: false,
            transitions,
        };

        return (
            <Sidebar {...sidebarProps}>
                <div className="top-area">
                    <MenuButton onClick={this.menuButtonClick} />
                    <Title text={this.findTitle()} />
                </div>
                <div className="content-area">
                    <Switch>
                        <Route exact path="/admin/dashboard" component={History} />
                        <Route exact path="/admin/clientes" component={Clients} />
                        <Route exact path="/admin/usuarios" component={Users} />
                        <Route exact path="/admin/maquinas" component={Machines} />
                        <Route exact path="/admin/formularios" component={Forms} />
                        <Route exact path="/admin/formularios/:id" component={Sections} />
                        <Route exact path="/admin/formularios/secciones/:id" component={Fields} />
                        <Route exact path="/entrada" component={CheckIn} />
                        <Route exact path="/salida" component={CheckOut} />
                    </Switch>
                </div>
            </Sidebar>
        );
    }
}

Main.contextType = StateContext;

Main.propTypes = {
    location: PropTypes.shape({
        pathname: PropTypes.string.isRequired,
    }),
};

Main.defaultProps = {
    location: {
        pathname: '',
    },
};

export default withRouter(Main);
