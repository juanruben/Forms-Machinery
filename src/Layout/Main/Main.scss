@import "../../styles.scss";

.top-area {
    background-color: $theme-background;
    padding: 0 20px;
    padding-top: 10px;
}

.content-area {
    background-color: $theme-background;
    min-height: calc(100vh - 150px);
    overflow: auto;
    padding: 20px;
    padding-top: 0px;
}
