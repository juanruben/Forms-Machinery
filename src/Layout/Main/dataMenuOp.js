export default [
    {
        id: 1,
        title: 'Dashboard',
        path: '/',
    },
    {
        id: 2,
        title: 'Registrar Entrada',
        path: '/entrada',
    },
    {
        id: 3,
        title: 'Registrar Salida',
        path: '/salida',
    },
];
