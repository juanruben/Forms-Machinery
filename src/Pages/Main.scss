@import "../Assets/css/general.scss";

.top-area {
  background-color: $icafal-background;
  padding: 0 20px;
  padding-top: 10px;
}

.content-area {
  background-color: $icafal-background;
  min-height: 100vh;
  overflow: auto;
  padding: 20px;
  padding-top: 0px;
}
