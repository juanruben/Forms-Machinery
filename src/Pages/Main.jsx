import React from 'react';
import PropTypes from 'prop-types';
import { Route, Switch, withRouter } from 'react-router-dom';
import Sidebar from 'react-sidebar';
// import { useStateValue } from '../State';
import Menu from '../Components/Menu/Menu';
import MenuButton from '../Components/MenuButton/MenuButton';
import Title from '../Components/Title/Title';
import Admin from './Admin';
import Operator from './Operator';

import dataMenu from './dataMenuAdmin';
import avatar from '../Assets/images/avatar.png';

import './Main.scss';

const mql = window.matchMedia('(min-width: 800px)');

class Main extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            docked: mql.matches,
            transitions: false,
        };

        this.menuButtonClick = this.menuButtonClick.bind(this);
        this.mediaQueryChanged = this.mediaQueryChanged.bind(this);
        this.findTitle = this.findTitle.bind(this);
    }

    componentDidMount() {
        mql.addListener(this.mediaQueryChanged);
        this.findTitle();
    }

    componentWillUnmount() {
        mql.removeListener(this.mediaQueryChanged);
    }

    mediaQueryChanged() {
        this.setState({
            docked: mql.matches,
        });
    }

    findTitle() {
        const { location } = this.props;
        const { pathname } = location;
        return dataMenu.find((item) => item.path === pathname).title;
    }

    menuButtonClick(event) {
        event.preventDefault();
        const { docked } = this.state;
        this.setState({
            transitions: true,
            docked: !docked,
        });
    }

    render() {
        const { docked, transitions } = this.state;

        const sidebarProps = {
            sidebar: <Menu items={dataMenu} username="administrador" avatar={avatar} />,
            docked,
            shadow: false,
            transitions,
        };

        return (
            <Sidebar {...sidebarProps}>
                <div className="top-area">
                    <MenuButton onClick={this.menuButtonClick} />
                    <Title text={this.findTitle()} />
                </div>
                <div className="content-area">
                    <Switch>
                        <Route exact path="/admin/dashboard" component={Admin} />
                        <Route exact path="/admin/clientes" component={Admin} />
                        <Route exact path="/admin/usuarios" component={Admin} />
                        <Route exact path="/admin/maquinas" component={Admin} />
                        <Route exact path="/admin/formularios" component={Admin} />
                        <Route exact path="/admin/dashboard" component={Admin} />
                        <Route exact path="/entrada" component={Operator} />
                        <Route exact path="/salida" component={Operator} />
                        <Route exact path="/dashboard" component={Operator} />
                    </Switch>
                </div>
            </Sidebar>
        );
    }
}

Main.propTypes = {
    location: PropTypes.shape({
        pathname: PropTypes.string.isRequired,
    }),
};

Main.defaultProps = {
    location: {
        pathname: '',
    },
};

export default withRouter(Main);
