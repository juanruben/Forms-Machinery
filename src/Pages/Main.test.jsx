import React from 'react';
import { shallow } from 'enzyme';
import Main from './Main';

test('should render ok', () => {
    const wrapper = shallow(<Main />);
    expect(wrapper.debug()).toMatchSnapshot();
});
