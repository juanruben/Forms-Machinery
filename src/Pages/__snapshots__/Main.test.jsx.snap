// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ok 1`] = `
"<Fragment>
  fjadslf:
  <button type=\\"submit\\" onClick={[Function: handleClick]}>
    Salir
  </button>
  <Switch>
    <Route path=\\"/admin\\" component={[function]} />
    <Route path=\\"/op\\" component={[Function: Operator]} />
    <Route component={[Function: Page404]} />
  </Switch>
</Fragment>"
`;
